### PR TITLE
 DBT-802 Add Materials to a lesson

### PR DESCRIPTION
**Pull request by kayiganwa04**
## Context

We are adding materials to lesson

## Test

- Login as admin
- got to manage lesson
- click one lesson and in the top click on lesson
- add a lesson/ delete and check the list of the lesson

## Platform
google chrome

### CODE REVIEW

File: src/modules/lessons/common/search-bar.vue [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    state() {
+      return this.$store.state[this.storeName]
+    }
+  },
+  mounted() {
+    this.loadWorkspaces()
+  },
+  methods: {
+    onChangeType(value) {
+      console.log('storename', this.storeName)
```
Comment (abelbordonado) [inline src/modules/lessons/common/search-bar.vue:96]: 🦞

File: src/modules/lessons/create-lesson/index.vue [hunk 2]
```diff
@@ -6,9 +6,9 @@
         <v-row dense :style="{ width: '-webkit-fill-available' }">
           <v-col cols="12" md="6">
             <FieldInput
-              id="name"
-              v-model="name"
-              label="Nombre de la Clase"
+              ref="lessonInput"
+              v-model="lesson.name"
+              label="Nombre del Clase"
```
Comment (abelbordonado) [inline src/modules/lessons/create-lesson/index.vue:11]: de la

File: src/modules/lessons/lesson-material-list/components/lesson-materials-list.vue [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+      <template v-slot:top>
+        <Toolbar title="Materiales" icon="mdi-folder-open">
+          <template slot="actions">
+            <resource-button icon-button="mdi-autorenew" @click="reset()" />
+          </template>
+        </Toolbar>
+        <SearchBar
+          :search-word="content"
+          :workspace="workspace"
+          :type="type"
```
Comment (abelbordonado) [inline src/modules/lessons/lesson-material-list/components/lesson-materials-list.vue:18]: Not needed, can be taken from the store directly :D  
But no need to change, if is already tested

File: src/modules/lessons/lesson-material-list/components/lesson-materials-list.vue [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+          <ResourceButtonAdd
+            ref="tagsInput"
+            text-button="Agregar"
+            class="mb-2 mx-3"
+            @click="onAddMaterial(item)"
+          />
+        </div>
+      </template>
+      <template v-slot:[`item.type`]="{ item }">
+        {{ MATERIAL_TYPES_LABELS[item.type] || 'aa' }}
```
Comment (abelbordonado) [inline src/modules/lessons/lesson-material-list/components/lesson-materials-list.vue:59]: `|| 'aa'` this seems not right, if the type is new we can say `Tipo Desconocido` or some other warning?

File: src/modules/lessons/lesson-material-list/components/lesson-materials-list.vue [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+        width: '400px',
+        icon: 'question',
+        title: 'ELIMINAR Material',
+        html: '<b>Esta acción es irreversible</b><br>¿Seguro que deseas eliminar este Material? Todos los materiales seran borrados del servidor y los alumnos no podrán acceder a ellos',
+        showConfirmButton: true,
+        showCancelButton: true,
+        confirmButtonColor: '#007bff',
+        confirmButtonText: 'Sí, eliminar',
+        cancelButtonText: 'Cancelar'
+      })
```
Comment (abelbordonado) [inline src/modules/lessons/lesson-material-list/components/lesson-materials-list.vue:196]: We shall make this into a function some were, that we can call statily

Toast.question({title , content}, {...options})

File: src/services/LessonRepository.js [hunk 6]
```diff
@@ -275,6 +275,85 @@ export default {
...
+      type,
+      tags,
+      orderBy,
+      order,
+      limit,
+      offset,
+      content: content || undefined
+    }
+
+    console.log('params', params)
```
Comment (abelbordonado) [inline src/services/LessonRepository.js:301]: 🦞

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
image

File: src/modules/lessons/lesson-material-list/components/lesson-materials-list.vue (hunk 5 above)
Comment (kayiganwa04) [reply src/modules/lessons/lesson-material-list/components/lesson-materials-list.vue:191]: later while refatoring right

Comment (abelbordonado) [reply src/modules/lessons/lesson-material-list/components/lesson-materials-list.vue]: We can do it later yes

**Comment by abelbordonado** [comment]
Lesson Materials
image

**Comment by abelbordonado** [comment]
image

**Comment by abelbordonado** [comment]
Fixed feedback on #20

**Pull request closed without merging**